src/expressions/operators/boolean/AndOperator.ts

```diff
@@ -11,6 +11,44 @@ import Expression from '../../Expression';
 import Specificity from '../../Specificity';
 import { DONE_TOKEN, ready } from '../../util/iterators';
 
+function intersectBuckets(bucket1: string | null, bucket2: string | null): string | null {
+	// null bucket applies to everything
+	if (bucket1 === null) {
+		return bucket2;
+	}
+	if (bucket2 === null) {
+		return bucket1;
+	}
+	// empty bucket applies to nothing
+	if (bucket1 === 'empty' || bucket2 === 'empty') {
+		return 'empty';
+	}
+	// Same bucket is same
+	if (bucket1 === bucket2) {
+		return bucket1;
+	}
+
+	if (bucket1.startsWith('name-')) {
+		// Name bucket always refers to an element or attribute
+		if (bucket2 === 'type-1' || bucket2 === 'type-1-or-type-2' || bucket2 === 'type-2') {
+			// A name is more specific than a type
+			return bucket1;
+		}
+		// Even if bucket2 is a name, we know it's not equal, so intersection is empty
+		return 'empty';
```

Comment:
This bucket is new. Won't we have any code in Fonto that will break because of it? In theory, other users can also use buckets (though I heavily doubt they do).

Also, when the bucket really _is_ `'empty'`, we can compile the `and` expression to a no-op that returns `false`, but that's for later.

Reply:
IIRC, previous problems were caused by new buckets having overlap with existing ones. Empty by definition doesn't overlap anything, so I'm not expecting problems. It will probably cause selectors that have this bucket to never be evaluated for retrieving metadata, which seems fine if they never match anyway.
In addition to a no-op, it might be worth warning in dev mode if this happens - any selector that doesn't match any nodes (aside from `false()`) is probably unintentional.